### PR TITLE
Add StartOnAddress function to listen on unix:// paths

### PR DESCRIPTION
**Pull request by jsafrane**

### CODE REVIEW

**Comment by jsafrane** [comment]
cc @pohly, @msau42

File: driver/mock.go [hunk 1]
```diff
@@ -61,6 +61,12 @@ func (m *MockCSIDriver) Start() error {
 	return nil
 }
 
+// StartOnAddress starts a new gRPC server listening on a random TCP loopback port.
+func (m *MockCSIDriver) Start() error {
```
Comment (pohly) [inline driver/mock.go]: Comment doesn't match the method.

I know it wasn't document before either (or at least not here), but when using random port allocation, how does the caller determine how to reach the mock driver, i.e. determine the port?

Comment (jsafrane) [reply driver/mock.go]: Good catch, comment is fixed now. Caller can call `MockCSIDriver.Address()` to get the actual address.

https://github.com/kubernetes-csi/external-attacher/blob/8f63c2c34b250355e73fa2d91e6cc65ea72f9645/pkg/connection/connection_test.go#L64-L66

**Comment by jsafrane** [comment]
@lpabon, can you please release new version of csi-test? It blocks https://github.com/kubernetes-csi/external-attacher/pull/123

**Comment by pohly** [comment]
I'd like to see PR #156 merged first. Then the new release becomes the first real-world test of image publishing with those build rules.

**Pull request merged**